astropy/io/fits/convenience.py

```diff
@@ -450,6 +450,15 @@ def writeto(
     checksum : bool, optional
         If `True`, adds both ``DATASUM`` and ``CHECKSUM`` cards to the
         headers of all HDU's written to the file.
+
+    Notes
+    -----
+    gzip and zip compression algorithms are natively supported.
+    Compression mode is automatically determined based on the filename extension
+    given ('.fits.gz' or '.fits.zip' respectively), or the compression used in a
+    pre-existing file that is being written to.
+    Note that bzip compression ('.fits.bz2') is also supported but requires
+    the ``bz2`` package.
```

Comment:
The availability of bz2 is a detail of the Python installation (if it was compiled with the bz2 lib), so not sure if we need to mention that. 
Suggested rewording:
```suggestion
    gzip, zip and bzip2 compression algorithms are natively supported.
    Compression mode is determined by the filename extension
    ('.gz', '.zip' or '.bz2' respectively).  It is also possible to pass a compressed
    file object, e.g. `gzip.GzipFile`.
```

Reply:
Thanks, I completely missed that !
I note that https://docs.python.org/3.10/library/bz2.html has no "availability" section, suggesting that it might be available everywhere now. Do you know in which situations we expect it to *not* be present ?

Reply:
Indeed, even Python docs does not mention it can be missing.
When you compile Python there are many optional libraries (bz2, zlib as well though it should be present on most systems, sqlite, tcl/tk etc.)